golangcli-lint latest fails

```
golangci-lint has version 1.44.0 built from 617470f on 2022-01-26T10:15:19Z
```

```
% make bin/pipelines-as-code lint test-unit
go build -mod=vendor -ldflags "-X github.com/openshift-pipelines/pipelines-as-code/pkg/params/version.Version="nightly-20220127-67c2cd4" "  -v -o bin/pipelines-as-code ./cmd/pipelines-as-code
Linting go files...
pkg/provider/bitbucketcloud/test/bbcloudtest.go:57:2: type assertion must be checked (forcetypeassert)
	prID := fmt.Sprintf("%d", event.Event.(*types.PullRequestEvent).PullRequest.ID)
	^
pkg/provider/bitbucketcloud/test/bbcloudtest.go:149:2: type assertion must be checked (forcetypeassert)
	prID := fmt.Sprintf("%d", event.Event.(*types.PullRequestEvent).PullRequest.ID)
	^
pkg/matcher/annotation_tasks_install.go:36:9: type assertion must be checked (forcetypeassert)
	return obj.(*tektonv1beta1.Task), nil
	      ^
pkg/provider/bitbucketserver/acl.go:55:3: type assertion must be checked (forcetypeassert)
		err := json.Unmarshal(comment.([]byte), activities)
		^
pkg/provider/bitbucketserver/pagination.go:46:6: type assertion must be checked (forcetypeassert)
		if isLastPage.(bool) {
		  ^
test/bitbucket_cloud_pullrequest_test.go:40:65: type assertion must be checked (forcetypeassert)
	checkSuccess(ctx, t, runcnx, opts, pullRequestEvent, targetNS, repobranch.Target["hash"].(string), title)
	                                                              ^
make: *** [Makefile:59: lint-go] Error 1
```

--- a/.tekton/pull-request.yaml
+++ b/.tekton/pull-request.yaml
@@ -93,7 +93,7 @@ spec:
                 make test GO_TEST_FLAGS="-v -race -coverprofile=coverage.txt -covermode=atomic"
 
             - name: lint
-              image: quay.io/app-sre/golangci-lint
+              image: quay.io/app-sre/golangci-lint:v1.44.0
               workingDir: $(workspaces.source.path)
               env:
                 - name: GOCACHE

--- a/pkg/matcher/annotation_tasks_install.go
+++ b/pkg/matcher/annotation_tasks_install.go
@@ -33,7 +33,12 @@ func (rt RemoteTasks) convertTotask(data string) (*tektonv1beta1.Task, error) {
 		return nil, fmt.Errorf("we have a task that is not looking like a kubernetes resource: task: %s resource: %w", data, err)
 	}
 
-	return obj.(*tektonv1beta1.Task), nil
+	task, ok := obj.(*tektonv1beta1.Task)
+	if !ok {
+		return nil, fmt.Errorf("this doesn't seem to be a proper task")
+	}
+
+	return task, nil
 }
 
 func (rt RemoteTasks) getTask(ctx context.Context, providerintf provider.Interface, task string) (*tektonv1beta1.Task, error) {

--- a/pkg/provider/bitbucketcloud/test/bbcloudtest.go
+++ b/pkg/provider/bitbucketcloud/test/bbcloudtest.go
@@ -54,7 +54,9 @@ func SetupBBCloudClient(t *testing.T) (*bitbucket.Client, *http.ServeMux, func()
 func MuxComments(t *testing.T, mux *http.ServeMux, event *info.Event, comments []types.Comment) {
 	assert.Assert(t, event.Event != nil)
 
-	prID := fmt.Sprintf("%d", event.Event.(*types.PullRequestEvent).PullRequest.ID)
+	pr, ok := event.Event.(*types.PullRequestEvent)
+	assert.Assert(t, ok)
+	prID := fmt.Sprintf("%d", pr.PullRequest.ID)
 	mux.HandleFunc("/repositories/"+event.Organization+"/"+event.Repository+"/pullrequests/"+prID+"/comments/",
 		func(rw http.ResponseWriter, r *http.Request) {
 			members := &types.Comments{
@@ -146,7 +148,9 @@ func MuxCreateCommitstatus(t *testing.T, mux *http.ServeMux, event *info.Event,
 func MuxCreateComment(t *testing.T, mux *http.ServeMux, event *info.Event,
 	expectedCommentSubstr string) {
 	assert.Assert(t, event.Event != nil)
-	prID := fmt.Sprintf("%d", event.Event.(*types.PullRequestEvent).PullRequest.ID)
+	prev, ok := event.Event.(*types.PullRequestEvent)
+	assert.Assert(t, ok)
+	prID := fmt.Sprintf("%d", prev.PullRequest.ID)
 
 	path := fmt.Sprintf("/repositories/%s/%s/pullrequests/%s/comments", event.Organization, event.Repository, prID)
 	mux.HandleFunc(path, func(rw http.ResponseWriter, r *http.Request) {

--- a/pkg/provider/bitbucketserver/acl.go
+++ b/pkg/provider/bitbucketserver/acl.go
@@ -52,7 +52,11 @@ func (v *Provider) checkOkToTestCommentFromApprovedMember(event *info.Event) (bo
 
 	for _, comment := range allPages {
 		activities := &activitiesTypes{}
-		err := json.Unmarshal(comment.([]byte), activities)
+		cbyte, ok := comment.([]byte)
+		if !ok {
+			return false, fmt.Errorf("cannot convert comment to bytes")
+		}
+		err := json.Unmarshal(cbyte, activities)
 		if err != nil {
 			return false, err
 		}

--- a/pkg/provider/bitbucketserver/pagination.go
+++ b/pkg/provider/bitbucketserver/pagination.go
@@ -43,7 +43,12 @@ func paginate(apiResultfunc apiResultfunc) ([]interface{}, error) {
 		if !ok {
 			break
 		}
-		if isLastPage.(bool) {
+		isLastPageb, ok := isLastPage.(bool)
+		if !ok {
+			break
+		}
+
+		if isLastPageb {
 			break
 		}
 	}

--- a/test/bitbucket_cloud_pullrequest_test.go
+++ b/test/bitbucket_cloud_pullrequest_test.go
@@ -37,7 +37,11 @@ func TestBitbucketCloudPullRequest(t *testing.T) {
 
 	pr, repobranch := createPR(t, bprovider, runcnx, bcrepo, opts, title, targetNS, targetRefName)
 	defer bitbucketTearDown(ctx, t, runcnx, bprovider, opts, pr.ID, targetRefName, targetNS)
-	checkSuccess(ctx, t, runcnx, opts, pullRequestEvent, targetNS, repobranch.Target["hash"].(string), title)
+
+	hash, ok := repobranch.Target["hash"].(string)
+	assert.Assert(t, ok)
+
+	checkSuccess(ctx, t, runcnx, opts, pullRequestEvent, targetNS, hash, title)
 }
 
 func createPR(t *testing.T, bprovider bitbucketcloud.Provider, runcnx *params.Run, bcrepo *bitbucket.Repository, opts E2EOptions, title, targetNS, targetRefName string) (*types.PullRequest, *bitbucket.RepositoryBranch) {

